truffle/src/main/java/org/jruby/truffle/translator/ParseEnvironment.java

```diff
@@ -40,14 +39,8 @@ public void popLexicalScope() {
         lexicalScope = lexicalScope.getParent();
     }
 
-    public long allocateReturnID() {
-        if (nextReturnID == Long.MAX_VALUE) {
-            throw new RuntimeException("Return IDs exhausted");
-        }
-
-        final long allocated = nextReturnID;
-        nextReturnID++;
-        return allocated;
+    public Object allocateReturnID() {
```

Comment:
Yeah I originally said Object, but you're right, we should have a class for it. So that for example it shows up well in allocation profiling.